### PR TITLE
test(SDK-3815) - Updates Unit Tests

### PR DESCRIPTION
**Pull request by Anush-Shand**
- Updates and Adds Unit Tests for 

Update tests in LocalDataStore.java
Update tests in AnalayticsManager.java
Update tests in ClevertapFactory.java
Add tests for ProfileValueHandler
Update tests in CryptUtils
Update tests in DBAdapter
Update tests in EventMediator
Update tests in EventQueueManager
Update tests in EvaluationManager
Update tests in EventAdapter
Update tests in TriggerAdapter
Update tests in TriggersMatcher
Update tests in LoginController
Update tests in Validator

### CODE REVIEW

File: clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapFactoryTest.kt [hunk 1]
```diff
@@ -203,4 +203,10 @@ class CleverTapFactoryTest : BaseTestCase() {
         val coreState = CleverTapFactory.getCoreState(application, cleverTapInstanceConfig, "12345")
         assertNotNull(coreState.cryptHandler)
     }
+
+    @Test
+    fun test_getCoreState_ReturnedObjectMustHaveNonNullProfileValueHandler(){
+        val coreState = CleverTapFactory.getCoreState(application, cleverTapInstanceConfig, "12345")
+        assertNotNull(coreState.profileValueHandler)
```
Comment (CTLalit) [inline clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapFactoryTest.kt:210]: assert that correct object is initialised

File: clevertap-core/src/test/java/com/clevertap/android/sdk/LocalDataStoreTest.kt [hunk 2]
```diff
@@ -130,26 +132,31 @@ class LocalDataStoreTest : BaseTestCase() {
...
+
+        assertNull(localDataStoreWithConfig.getProfileProperty("key1"))
+    }
+
 
+    @Test
+    fun test_getProfileProperty_when_FunctionIsCalledWithCorrectKeyAndEncryptedValue_should_ReturnNull() {
+        localDataStoreWithConfig.updateProfileFields(mapOf("key" to "[abcd]"))
+
+        assertNull(localDataStoreWithConfig.getProfileProperty("key"))
```
Comment (CTLalit) [inline clevertap-core/src/test/java/com/clevertap/android/sdk/LocalDataStoreTest.kt:159]: assert the value as well, 
localDataStoreWithConfig.getProfileProperty("key") == "[abcd]"

Comment (Anush-Shand) [reply clevertap-core/src/test/java/com/clevertap/android/sdk/LocalDataStoreTest.kt:159]: getProfileProperty will return null as the value is encrypted

File: clevertap-core/src/test/java/com/clevertap/android/sdk/AnalyticsManagerTest.kt
Comment (CTLalit) [inline clevertap-core/src/test/java/com/clevertap/android/sdk/AnalyticsManagerTest.kt:1]: All the actions/method calls/things to test are made inside mockStatic block which we can separate out.

File: clevertap-core/src/test/java/com/clevertap/android/sdk/EventQueueManagerTest.kt [hunk 3]
```diff
@@ -132,21 +133,53 @@ class EventQueueManagerTest : BaseTestCase() {
...
+
+    @Test
+    fun test_queueEvent_for_profileEvent_will_process_further_and_add_to_queue_when_event_should_not_be_dropped() {
+        mockStatic(CTExecutorFactory::class.java).use {
+            `when`(CTExecutorFactory.executors(cleverTapInstanceConfig)).thenReturn(
+                MockCTExecutors(
+                    cleverTapInstanceConfig
+                )
+            )
+            val captor = ArgumentCaptor.forClass(Runnable::class.java)
```
Comment (CTLalit) [inline clevertap-core/src/test/java/com/clevertap/android/sdk/EventQueueManagerTest.kt:165]: unused

**Review by CTLalit — APPROVED** [review]
Approved with minor comments.

Investigate if parameterised tests are possible for event*manager, mediator

**Pull request merged**